.circleci/config.yml

```diff
@@ -92,385 +108,894 @@ commands:
+            gcloud container clusters list --filter="name=<< parameters.cluster >>" --format="value(name)"
+            if [ $? -eq 0 ] && [ $cluster_exists -eq 1 ]; then
+              gcloud container clusters delete << parameters.cluster >> --quiet
+            fi
+            exit 0
+      - run:
+          name: "Create GKE cluster"
+          command: |
+            gcloud container clusters create $GKE_CLUSTER_NAME \
+            --num-nodes=<< parameters.num_nodes >> --machine-type=<< parameters.machine_type >> \
+            --disk-size=<< parameters.disk_size >> --disk-type=<< parameters.disk_type >> \
+            --accelerator=type=<< parameters.gpu_type >>,count=<< parameters.gpu_count >>
+#            list gpus available in zone:
+#            gcloud compute accelerator-types list | grep us-central1
+#            TPU support
+#           gcloud beta container clusters update cluster-name --enable-tpu
+#           gcloud container clusters get-credentials
+#            TPU support
+#            --release-channel=stable \
+#            --scopes=cloud-platform \
+#            --enable-ip-alias \
+#            --enable-tpu
+#            e2-medium
+#            --accelerator=type=nvidia-tesla-k80,count=1
+          environment:
+            GKE_CLUSTER_NAME: << parameters.cluster >>
+          no_output_timeout: << parameters.no-output-timeout>>
+
+  get_gke_cluster_credentials:
+    parameters:
+      cluster:
+        description: "cluster name"
+        type: string
+        default: << pipeline.parameters.gcp_cluster_name >>
+    steps:
+      - run:
+          name: "Get gke cluster credentials"
+          command: |
+            gcloud components install gke-gcloud-auth-plugin
+            gcloud container clusters get-credentials << parameters.cluster >>
+          environment:
+            GKE_CLUSTER_NAME: << parameters.cluster >>
+
+jobs:
+  test:
+    resource_class: xlarge
+    parameters:
+      image:
+        type: string
+      toxenv:
+        type: string
+      parallelism:
+        type: integer
+        default: 1
+      xdist:
+        type: integer
+        default: 1
+      notify_on_failure:
+        type: boolean
+        default: false
+    docker:
+      - image: << parameters.image >>
+    working_directory: /mnt/ramdisk
+    steps:
+      - checkout
+      - run:
+          name: Install system deps
+          command: apt-get update && apt-get install -y libsndfile1 ffmpeg
+      - run:
+          name: Install python dependencies
+          command: |
+            pip install tox==<< pipeline.parameters.tox_version >>
+      - restore-tox-cache
+      - run:
+          name: Run tests
+          command: |
+            tox -v -e << parameters.toxenv >>
+          no_output_timeout: 25m
+      - save-tox-cache
+      - save-test-results
+      # conditionally post a notification to slack if the job failed
```

Comment:
why not publish also on success? If someone wants to look at the completed run?

Reply:
yea, good idea! will add